src/Components/Components/src/Auth/AuthorizeViewCore.cs

```diff
@@ -102,10 +102,27 @@ protected override async Task OnParametersSetAsync()
         private async Task<bool> IsAuthorizedAsync(ClaimsPrincipal user)
         {
             var authorizeData = GetAuthorizeData();
+            EnsureNoAuthenticationSchemeSpecified(authorizeData);
+
             var policy = await AuthorizationPolicy.CombineAsync(
                 AuthorizationPolicyProvider, authorizeData);
             var result = await AuthorizationService.AuthorizeAsync(user, Resource, policy);
             return result.Succeeded;
         }
+
+        private static void EnsureNoAuthenticationSchemeSpecified(IAuthorizeData[] authorizeData)
+        {
+            // It's not meaningful to specify a nonempty scheme, since by the time Components
+            // authorization runs, we already have a specific ClaimsPrincipal (we're stateful).
+            // To avoid any confusion, ensure the developer isn't trying to specify a scheme.
+            for (var i = 0; i < authorizeData.Length; i++)
+            {
+                var entry = authorizeData[i];
+                if (!string.IsNullOrEmpty(entry.AuthenticationSchemes))
+                {
+                    throw new NotSupportedException($"The authorization data specifies an authentication scheme with value '{entry.AuthenticationSchemes}'. Authentication schemes cannot be specified for components.");
+                }
+            }
+        }
```

Comment:
But you could have multiple schemes and have the scheme act as a filter (that's how I recall it works today). Don't we want to be consistent with the rest of the platform?

Reply:
The schemes are passed in to the authentication system when used with server-side auth: https://github.com/aspnet/AspNetCore/blob/master/src/Security/Authorization/Policy/src/PolicyEvaluator.cs#L35

Given that server-side auth with have already happened at this point, there isn't really anything we can do here that would be consistent.

 yes, you could filter but that not going to work with a passive scheme like bearer.

Since we can't be consistent, and we don't have a clear use case, it seems better to just block it.

Reply:
It's not supported at the `IAuthenticationStateProvider` layer, as that only has an API to get the "current authentication state" which is a `ClaimsPrincipal`. It doesn't have a way to get different principals for different schemes.

In the future we could choose to make a more advanced version of `IAuthenticationStateProvider` (e.g., `ISchemeAuthenticationStateProvider`) that has APIs for getting different authentication states for different schemes. If we do that then we could relax this limitation. But since we're not planning that for 3.0, this "fail hard" rule is to help people not get confused about the features set.

Reply:
I guess we can do this in the future without breaking changes.